persistence/model/src/main/java/gov/va/vro/persistence/repository/ClaimSubmissionRepository.java

```diff
@@ -0,0 +1,22 @@
+package gov.va.vro.persistence.repository;
+
+import gov.va.vro.persistence.model.ClaimSubmissionEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Repository
+public interface ClaimSubmissionRepository extends JpaRepository<ClaimSubmissionEntity, UUID> {
+
+  List<ClaimSubmissionEntity> findByReferenceIdAndIdType(String referenceId, String idType);
+
+  // For the few cases where we really don't know an idType, this is equivalent to the prior Claim
+  // find byReferenceId not specifying a type.
+  Optional<ClaimSubmissionEntity> findFirstByReferenceIdOrderByCreatedAtDesc(String referenceId);
```

Comment:
Reminder to self: what value of idType is being used when only the referenceId is provided?

Reply:
Re: the comment above for v2 types and this it does look like the routes aren't setting the type correctly, and I will make them set the type to "MAS". 

As far as removing this method, i left it in place for claim metrics since that is a v1 api and it can only specify the reference id but not the id type unless I change the api contract. That is the only place it's used. If that's not what we want, I can certainly do any of the following:
1. change the claim metrics API point to ask for the id type
2. change the claim metrics API point to only search one type or the other (presumably the v1 type). 
3. Leave it, with the slight risk of this method's existence, but note that it is only called and should only ever be called by the metrics api.

Reply:
I prefer bullet 1 if it's not too burdensome. The Claim Metrics API is actually relevant to both v1 and v2, but it's not a critical API and only used for our reporting purposes.
Alternatively, you can do bullet 2 using `idType="MAS"` (for v2 endpoints) -- I know, a bit confusing (sorry).